katsdptelstate/rdb_utility.py

```diff
@@ -98,29 +157,32 @@ def dump_zset(data):
     if entry_count > 127 or sum(len(entry) for entry in data) >= 2**31:
         # for inscrutable reasons the length here is half the number of actual
         # entries (i.e. scores are ignored)
-        enc = [b'\x03' + encode_len(len(data))]
-        for entry in data:
+        return dump_iterable(
```

Comment:
I knew you wouldn't be able to resist rewriting this at some stage - basically all the code I ever write is eventually Bruce'd :)